conan/tools/cmake/cmakedeps/cmakedeps.py

```diff
@@ -32,6 +33,7 @@ def __init__(self, conanfile):
 
         # Enable/Disable checking if a component target exists or not
         self.check_components_exist = False
+        self._properties = defaultdict(dict)
```

Comment:
I don't love the defaultdict, because it can easily create entries for non-existing dependencies, which is valid python, but non logical model. I typically prefer to resort to regular dict and the idiom ``mydict.setdefault(key, default_value).update...``